scalecube-configuration/src/main/java/io/scalecube/configuration/ConfigurationServiceImpl.java

```diff
@@ -1,133 +1,100 @@
-import io.scalecube.configuration.tokens.TokenVerifier;
-import java.util.Objects;
+import io.scalecube.configuration.repository.ConfigurationRepository;
+import io.scalecube.configuration.repository.Document;
+import io.scalecube.configuration.repository.Repository;
+import io.scalecube.security.api.AccessControl;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
-import reactor.core.scheduler.Schedulers;
 
 public class ConfigurationServiceImpl implements ConfigurationService {
 
   private static final Logger logger = LoggerFactory.getLogger(ConfigurationServiceImpl.class);
+  private static final Acknowledgment ACK = new Acknowledgment();
 
-  private final ConfigurationDataAccess dataAccess;
-  private TokenVerifier tokenVerifier;
-  private final AuthorizationService authorizationService;
+  private final ConfigurationRepository repository;
+  private final AccessControl accessControl;
 
-  private ConfigurationServiceImpl(
-      ConfigurationDataAccess dataAccess, TokenVerifier tokenVerifier) {
-    this.dataAccess = dataAccess;
-    this.tokenVerifier = tokenVerifier;
-    this.authorizationService = AuthorizationService.builder().build();
+  public ConfigurationServiceImpl(ConfigurationRepository repository, AccessControl accessContorl) {
+    this.repository = repository;
+    this.accessControl = accessContorl;
   }
 
   @Override
   public Mono<Acknowledgment> createRepository(CreateRepositoryRequest request) {
-    return Mono.fromRunnable(() -> logger.debug("createRepository: enter: request: {}", request))
-        .then(Mono.fromCallable(ServiceOperationFactory::createRepository))
-        .flatMap(operation -> Mono
-            .from(operation.execute(request, context(OperationType.CreateRepoitory))))
-        .subscribeOn(Schedulers.parallel())
-        .doOnSuccess(result -> logger.debug("createRepository: exit: request: {}", request))
-        .doOnError(th -> logger.error("createRepository: request: {}, error: {}", request, th));
+    verifyRequest(request);
+    return accessControl
+        .check(request.token().toString(), ConfigurationService.CONFIG_CREATE_REPO)
+        .flatMap(
+            p -> this.repository.createRepository(new Repository(p.tenant(), request.repository())))
+        .map(b -> ACK);
   }
 
   @Override
   public Mono<FetchResponse> fetch(FetchRequest request) {
-    return Mono.fromRunnable(() -> logger.debug("fetch: enter: request: {}", request))
-        .then(Mono.fromCallable(ServiceOperationFactory::fetch))
-        .flatMap(operation -> Mono.from(operation.execute(request, context(OperationType.Read))))
-        .subscribeOn(Schedulers.parallel())
-        .doOnSuccess(
-            response -> logger.debug("fetch: exit: request: {}, response: {}", request, response))
-        .doOnError(th -> logger.error("fetch: request: {}, error: {}", request, th));
+    verifyRequest(request);
```

Comment:
whydon't you want to make `verifyRequest` to return mono and chain it nicely with all the rest of the calls?